mlflow/langchain/runnables.py

```diff
@@ -0,0 +1,282 @@
+    from langchain.chains.loading import type_to_loader_dict as chains_type_to_loader_dict
+    from langchain.llms.loading import get_type_to_cls_dict as llms_get_type_to_cls_dict
+    from langchain.llms.loading import load_llm
+    from langchain.prompts.loading import load_prompt
+
+    config_path = os.path.join(path, model_config.get(_MODEL_DATA_KEY, _MODEL_DATA_YAML_FILE_NAME))
+    # Load runnables from config file
+    if config_path.endswith(".yaml"):
+        config = _load_from_yaml(config_path)
+    elif config_path.endswith(".json"):
+        config = _load_from_json(config_path)
+    else:
+        raise MlflowException(
+            f"Cannot load runnable without a config file. Got path {config_path!s}."
+        )
+    _type = config.get("_type")
+    if _type in chains_type_to_loader_dict:
+        return load_chain(config_path)
+    elif _type in prompts_types:
+        return load_prompt(config_path)
+    elif _type in llms_get_type_to_cls_dict():
+        return load_llm(config_path)
+    elif _type in custom_type_to_loader_dict():
+        return custom_type_to_loader_dict()[_type](config)
+    raise MlflowException(f"Unsupported type {_type} for loading.")
+
+
+def _load_model_from_path(path: str, model_config=None):
+    model_load_fn = model_config.get(_MODEL_LOAD_KEY)
+    if model_load_fn == _RUNNABLE_LOAD_KEY:
+        return _load_runnables(path, model_config)
+    if model_load_fn == _BASE_LOAD_KEY:
+        return _load_base_lcs(path, model_config)
+    if model_load_fn == _CONFIG_LOAD_KEY:
+        return _load_model_from_config(path, model_config)
+    raise MlflowException(f"Unsupported model load key {model_load_fn}")
+
+
+def _load_runnable_with_steps(file_path: Union[Path, str], model_type: str):
+    """
+    Load the model
+
+    :param file_path: Path to file to load the model from.
+    :param model_type: Type of the model to load.
+    """
+    from langchain.schema.runnable import RunnableParallel, RunnableSequence
+
+    # Convert file to Path object.
+    load_path = Path(file_path) if isinstance(file_path, str) else file_path
+    if not load_path.exists() or not load_path.is_dir():
+        raise MlflowException(
+            f"File {load_path!s} must exist and must be a directory "
+            "in order to load runnable with steps."
+        )
+
+    steps_conf_file = load_path / _RUNNABLE_STEPS_FILE_NAME
+    if not steps_conf_file.exists():
+        raise MlflowException(
+            f"File {steps_conf_file} must exist in order to load runnable with steps."
+        )
+    steps_conf = _load_from_yaml(steps_conf_file)
+    steps_path = load_path / _STEPS_FOLDER_NAME
+    if not steps_path.exists() or not steps_path.is_dir():
+        raise MlflowException(
+            f"Folder {steps_path} must exist and must be a directory "
+            "in order to load runnable with steps."
+        )
+    if model_type == RunnableSequence.__name__:
+        steps = []
```

Comment:
Is it possible to always use a dict? For `RunnableSequence`, we can sort values by key.